internal/proxy/task_query.go

```diff
@@ -240,6 +240,28 @@ func (t *queryTask) createPlan(ctx context.Context) error {
 	return nil
 }
 
+func (t *queryTask) CanSkipAllocTimestamp() (bool, error) {
+	collID, err := globalMetaCache.GetCollectionID(context.Background(), t.request.GetDbName(), t.request.GetCollectionName())
+	if err != nil { // err is not nil if collection not exists
+		return false, err
+	}
+
+	collectionInfo, err2 := globalMetaCache.GetCollectionInfo(context.Background(), t.request.GetDbName(), t.request.GetCollectionName(), collID)
+	if err2 != nil {
+		return false, err2
+	}
+
+	var consistencyLevel commonpb.ConsistencyLevel
+	useDefaultConsistency := t.request.GetUseDefaultConsistency()
+	if useDefaultConsistency {
+		consistencyLevel = collectionInfo.consistencyLevel
+	} else {
+		consistencyLevel = t.request.GetConsistencyLevel()
```

Comment:
I think this check should be moved above, which may also skip to get collection schema from rootcoord.